scene/gui/label.cpp

```diff
@@ -206,6 +206,9 @@ void Label::_shape() {
 			case TextServer::OVERRUN_NO_TRIMMING:
 				break;
 		}
+		if (horizontal_alignment == HORIZONTAL_ALIGNMENT_RIGHT) {
+			overrun_flags.set_flag(TextServer::OVERRUN_ELLIPSIS_AT_START);
+		}
```

Comment:
Is this needed here? Can't you set `label.text_overrun_behavior = TextServer.OVERRUN_ELLIPSIS_AT_START` manually? If yes, this feels like unnecessary hardcoding.

I think it's also wrong for RTL, because H Align = Right *always* means the right hand side whereas `OVERRUN_ELLIPSIS_AT_START` rightfully depends on the text direction.

Reply:
> because H Align = Right always means

`H_ALIGN_RIGHT` is left in RTL layout direction (not base text direction, it's not related to alignment in any way).

Reply:
But it's likely should be controllable.

Reply:
You are setting `text_direction` (which is almost always should be set to `Auto`, it's only used to force direction for language neutral text), what's used when locale is set to the RTL language is `layout_direction` (under Control → Layout).

Reply:
That makes more sense now. Thanks for explaining!